src/runtime.rs

```diff
@@ -276,6 +276,10 @@ where
 
             local_state.query_stack.pop().unwrap()
         };
+        // Transitively propagete untracked.
+        if subqueries.is_none() {
```

Comment:
This is a crude fix. I *think* it makes cancellation correct, but it also causes unnecessary recomputations if volatile queries are involved. 

I believe that with just volatile queries and no cancellation the previous logic was correct. What makes cancellation special is the fact that "are we going to do an untracked read?" is non-deterministic. With volatile, it **is** deternenistic and depends on the previous queries. So, if you swtich from "has untracked read" to "no untracked read", that means that some your dependency bumped changed_at, and so the monotonicity of changed_at is maintained.